src/rpc/blockchain.cpp

```diff
@@ -273,6 +273,7 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* blockindex, bool tx
     valuePools.push_back(ValuePoolDesc("sprout", blockindex->nChainSproutValue, blockindex->nSproutValue));
     valuePools.push_back(ValuePoolDesc("sapling", blockindex->nChainSaplingValue, blockindex->nSaplingValue));
     valuePools.push_back(ValuePoolDesc("orchard", blockindex->nChainOrchardValue, blockindex->nOrchardValue));
+    valuePools.push_back(ValuePoolDesc("lockbox", blockindex->nChainLockboxValue, blockindex->nLockboxValue));
```

Comment:
Hmm, I was originally medium-NACK on this but now am undecided. `lockbox` is not a value pool in the same sense as the others, and I think including it in `valuePools` in JSON-RPC output could be misleading.

However, I believe there is a bug in the PR, as it doesn't alter this check:
https://github.com/zcash/zcash/blob/98495e842ea592593f8fdb0ecf49252cf77b7b85/src/main.cpp#L3585-L3603

and I think that is because currently `nChainTotalSupply` is modified by adding `coinbaseTx.GetValueOut() - sum(nonCoinbaseTx fees)`, which post-NU6 will not include the lockbox funds (as not all funding streams will have coinbase outputs). This currently does not break the above check, because the lockbox funds are omitted from both the chain total supply, and from the set of value pools that the check enforces against.

So I think we have two options, and we absolutely need to be consistent with `zebrad` here:
- Leave `nChainTotalSupply` defined as-is (only counting non-lockbox funding stream outputs).
  - This has the effect that the lockbox funds are considered "unissued" and do not count towards the total issued supply (altering the issuance curve). It would also unambiguously mean that `lockbox` is _not_ a value pool.
- Fix the implementation of `nChainTotalSupply` to count _all_ funding stream outputs, and then add `nChainLockboxValue` to the other side of the check.
  - This has the effect that the lockbox funds are considered "issued" and count towards the total issued supply (leaving the issuance curve unaltered), and just not tied to a spendable UTXO. It would also mean that `lockbox` _might_ be considered a value pool (as it is "a pool where issued funds exist, with mechanism for spending", just with "mechansim" not yet defined).

Reply:
I think that the correct behavior here is to fix `nChainTotalSupply`; we have defined the lockbox as a pool of _issued_ funds, therefore it should be considered part of the supply (albeit currently not supply that is able to circulate.) This is consistent with what other chains have done with respect to lockup periods; people hold tokens, but those tokens cannot be transferred until the lockup period is over.

Reply:
Fixed in 92616d94bf83e82d58afc76a6d462a319c8c3994